src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/SpecimenMapper.java

```diff
@@ -33,10 +33,11 @@
     private final DateFormatMapper dateFormatMapper;
     private final ResourceFullUrlGenerator fullUrlGenerator;
 
-    public List<Specimen> mapToSpecimens(final Message message, Patient patient) {
+    public Map<String, Specimen> mapToSpecimens(final Message message, Patient patient) {
```

Comment:
This is no longer obvious what is being returned. It's a map of the edifact's SEQ numbers for each specimen, pointing to the fhir Specimen itself (because the latter doesn't contain the former, but we still need to know it later).

Some options could be
1. rename the method to something that makes it obvious - suggestions welcome
```suggestion
    public Map<String, Specimen> mapToSpecimensBySequenceNumber(final Message message, Patient patient) {
```
2. create our own extension of `HashMap` with a suitable name that defines the generics already and return that instead
```suggestion
    public SpecimensBySequenceNumber mapToSpecimens(final Message message, Patient patient) {
```
```
class SpecimensBySequenceNumber extends HashMap<String, Specimen> {}
```
3. use the (required) `SequenceDetails` edifact segment class as the key instead of its `number` field
```suggestion
    public Map<SequenceDetails, Specimen> mapToSpecimens(final Message message, Patient patient) {
```

Reply:
Personally, I would vote for option 1 because I think it will be clear enough in this case. Option 3 is also good and clear. Option 2 is of course also clear but I personally think that it is not necessary to create our own extension of a HashMap when we have much easier to implement alternative solutions

Reply:
If we want to keep the `String` type in the map, then I'm in favour of Option 1, but if we want to include the `SequenceDetails` then the third option would be better.  
My own preference is Option 1 for simplicity because I think the name of the variable makes it clear what the String key of the Map is. 


I'm not a fan of Option 2 because I don't think we need a new class for something that can be held by a Map data type.